### PR TITLE
GROOVY-7416: Problem With CompileStatic And Interface Methods In Abst…

### PR DESCRIPTION
**Pull request by paulk-asert**
…ract Subclass

### CODE REVIEW

**Comment by blackdrag** [comment]
I have the feeling that this code will still ignore methods coming from an interface that has been extended, especially if there are multiple such interfaces.  

    interface SomeInterface {void foo()}
    interface AnotherInterface extends SomeInterface;
    abstract class Parent implements AnotherInterface;
    abstract abstract class Child extends Parent{ void bar(){foo()}}

**Comment by paulk-asert** [comment]
It seems to be okay. I added another testcase but perhaps I didn't quite catch the case you were after?

File: src/test/groovy/transform/stc/BugsSTCTest.groovy [hunk 1]
```diff
@@ -650,4 +650,42 @@ Printer
...
+        assertScript '''
+            import groovy.transform.CompileStatic
+
+            interface SomeInterface {
+                void someInterfaceMethod()
+            }
+
+            abstract class AbstractSuperClass implements SomeInterface {}
+
+            @CompileStatic
```
Comment (shils) [inline src/test/groovy/transform/stc/BugsSTCTest.groovy]: Are the CompileStatic annotations here and in the following test necessary? BugsSTCTest extends StaticTypeCheckingTestCase.

**Comment by paulk-asert** [comment]
They can be removed.

**Comment by paulk-asert** [comment]
merged

**Pull request closed without merging**